### ISSUE TITLE
Assertion fail in DiscretePD::sample(blacklist)

### ISSUE THREAD

**Issue by geisserf** [prost-planner/prost#80]
When running the planner in debug mode in academic_advising_02 with `./prost.py --debug academic-advising_inst_mdp__02 [PROST -se [THTS -act [UCB1] -out [UMC] -backup [PB] -init [Expand -h [IDS]]]]` the following assertion is violated in the last planning step:

```
search-debug: /home/admin-u1064334/git/prost/src/search/probability_distribution.cc:61:
std::pair<double, double> DiscretePD::sample(const std::vector<int>&) const: Assertion `false' failed.
```

**Comment by geisserf** [comment]
The issue is that the call `    double randNum = MathUtils::rnd->genDouble(0.0, remainingProbSum);` with remainingProbSum = 0.09 returns 0.2, which should not happen.

Since this concerns the RNG it might also be an issue specific to my machine, I use gcc 7.4.0.

**Comment by geisserf** [comment]
The problem appears only in the debug version and does not affect the release version. The reason is that in the unit test probability_distribution_test.cc the standard random number generator (RNG) is exchanged for a pseudo RNG. Since the pointer to the RNG in math_utils.h is static and unit tests are executed before any part of search starts the unit test overwrites the RNG that is used within search.

A simple solution is of course to not run unit tests before search, even in the debug version. Since we already run unit tests when building the planner we might remove them from the debug version as well, to ensure that they do not interfere with search under any circumstance.

**Comment by geisserf** [comment]
To make this more explicit: right now in debug version whenever a random number between [min,max] is requested the following code is executed:
```
    // Alternate generated numbers
    double genDouble(double, double) override {
        ++counter;
        if (counter % 2) {
            return 0.2;
        } else {
            return 0.4;
        }
    }
```

**Comment by thomaskeller79** [comment]
This issue has been resolved in fcb229fda1e21cb75fe7bea75d18c3b2e2dc3840
